core/include/scipp/core/time_point.h

```diff
@@ -4,36 +4,74 @@
 #pragma once
 
 #include <stdint.h>
+#include <string>
 
 namespace scipp::core {
 
 /// Time-point similar to std::chrono::time point but without compile-time unit.
+/// The unit is determined by the Variable the time_point is stored in.
 class time_point {
 public:
-  constexpr time_point() noexcept = default;
-  constexpr time_point(const int64_t &d) noexcept : m_duration(d){};
-  constexpr time_point(const int32_t &d) noexcept
-      : m_duration(static_cast<int64_t>(d)){};
+  time_point() = default;
+  explicit time_point(int64_t duration) : m_duration{duration} {}
+  time_point(const time_point &other) = default;
+  time_point(time_point &&other) noexcept = default;
+  time_point &operator=(const time_point &other) = default;
+  time_point &operator=(time_point &&other) noexcept = default;
+  ~time_point() noexcept = default;
```

Comment:
Why are these required explicitly? What is preventing compiler generation?

Reply:
Nothing is preventing it. Since there is a non-default constructor, I had to add the default ctor. And I just followed the advice of write one or none.
Should I rather omit these?

Reply:
Not sure I saw the same advice?

Reply:
Hm, maybe just a question of style? Removed it now. I don't really care about it either way...

Reply:
Used to be "rule of 3" before C++11, then "rule of 5 but actually usually 0". See, e.g.,:
https://en.cppreference.com/w/cpp/language/rule_of_three